ros/lib/utils.py

```diff
@@ -185,3 +193,61 @@ def log_request(self, code='-', size='-'):
             return
         else:
             super().log_request(code, size)
+
+
+def generate_highlight_description(instance_type):
+    if instance_type not in INSTANCE_TYPES.keys():
+        # logging lookup failure on Prometheus
+        ec2_instance_lookup_failures.labels(reporter='API Events').inc()
+        return 'NA'
+
+    cpu_type = INSTANCE_TYPES[instance_type]['extra']['physicalProcessor']
+    num_vcpus = INSTANCE_TYPES[instance_type]['extra']['vcpu']
+    ram_gb = INSTANCE_TYPES[instance_type]['extra']['memory']
+    cloud_regions = INSTANCE_TYPES[instance_type]['extra']['regionCode']
+    cloud = 'AWS'  # Update when multi cloud provider support is added
+
+    description_text = f'{cpu_type} instance with {num_vcpus} vCPUs ' \
+                       f'and {ram_gb} of RAM, running on {cloud} ' \
+                       f'{cloud_regions} regions'
+
+    return description_text
+
+
+def highlights_instance_types(queryset, highlight_type):
+    instance_candidates, values_dict, highlights_list = [], {}, []
+    if highlight_type == 'current':
+        for performance_profile in queryset:
+            try:
+                instance_candidates.append(performance_profile.rule_hit_details[0]['details']['instance_type'])
+            except (IndexError, KeyError):
+                # Systems w/o profiles are being considered in 'current'
+                # Ex: Optimized, Waiting for Data, etc.
+                system_record = db.session.query(System).\
+                    filter_by(id=performance_profile.system_id).first()
+                if system_record:
+                    instance_candidates.append(system_record.instance_type)
```

Comment:
@saltgen  - instead of special handling for keyerror/indexerror, will it be good idea to join performance_profile and system table? 

IIUC, mainly it requires inventory_id, system.state, system.instance_type, system.region, rule_hit_details.
Will it make more sense to load required fields than all fields details?

Reply:
Change pushed